### PR TITLE
Diamond Hoe

### PR DESCRIPTION
**Pull request by Yeah-I-listening-Hollywood-undead**
## About the PR
Added Diamond Hoe for very successful gardeners

## Why / Balance

This will not affect the game balance in any way, since this hoe does not differ from the standard one in anything other than its appearance. That make the reference to minecraft.
## Technical details

At the moment, crafting this hoe causes the server to explode. I do not know how to fix it, I hope you will help
## Media

Screenshot_1058

- [✔] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

A diamond hoe has been added. You can make it out of two diamonds and two sticks. And extremely upset their fellow miners who risked their lives for these diamonds.

:cl:
- add: Diamond hoe and craft for diamond hoe!

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml [hunk 1]
```diff
@@ -20,6 +20,28 @@
   - type: Item
     sprite: Objects/Tools/Hydroponics/hoe.rsi
 
+- type: entity
+  name: Diamond hoe
```
Comment (UbaserB) [inline Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml:24]: make it lowercase

**Review by TheShuEd — CHANGES_REQUESTED** [review]

**Comment by TheShuEd** [comment]
let the unit tests do their job.

**Comment by lzk228** [comment]
also your cl is broken, should be 
```
:cl:
- tweak: text
- add: text
```

File: Resources/Prototypes/Recipes/Crafting/Graphs/improvised/diamond_hoe.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  start: start
+  graph:
+  - node: start
+    edges:
+    - to: DiamondHoe
+      steps:
+      - material: WoodPlank
+        amount: 2
+        doAfter: 10
+      - material: MaterialDiamond
```
Comment (TheShuEd) [inline Resources/Prototypes/Recipes/Crafting/Graphs/improvised/diamond_hoe.yml:12]:
```suggestion
      - material: RawDiamond
```

**Comment by regenpower** [comment]
this is why i live for this game

**Comment by BombasterDS** [comment]
RawDiamonds is now stackable thing, try use Diamonds instead in craft recipe

**Comment by BombasterDS** [comment]
Actually, I think it's better not to be craftable, but a loadout equipment for longtime botany players (like janitor's golden plunger)

File: Resources/Prototypes/Recipes/Crafting/Graphs/improvised/diamond_hoe.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  start: start
+  graph:
+  - node: start
+    edges:
+    - to: DiamondHoe
+      steps:
+      - material: WoodPlank
+        amount: 2
+        doAfter: 10
+      - material: RawDiamond
```
Comment (BombasterDS) [inline Resources/Prototypes/Recipes/Crafting/Graphs/improvised/diamond_hoe.yml:12]: Try use Diamond instead of RawDiamond

**Comment by SlamBamActionman** [comment]
Can we at least make it colored the same as the diamond item, to make it more cohesive and less of a blatant Minecraft reference than what it already is?

**Comment by BombasterDS** [comment]
You need to add ConstructionComponent to it, as failed test says

**Comment by Yeah-I-listening-Hollywood-undead** [comment]
> You need to add ConstructionComponent to it, as failed test says

I don't see  ConstructionComponent in any YML prototypes

**Comment by BombasterDS** [comment]
> > You need to add ConstructionComponent to it, as failed test says
> 
> I don't see ConstructionComponent in any YML prototypes

Add
- type: Construction
  graph: DiamondHoe
  node: DiamondHoe (actually, nodes must be named lowercased, so it's better to call node "hoe" there and in construction graph)

File: Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml [hunk 4]
```diff
@@ -20,6 +20,28 @@
...
+    state: icon
+  - type: MeleeWeapon
+    wideAnimationRotation: 135
+    swingLeft: true
+    damage:
+      types:
+        Blunt: 10
+  - type: Item
+    sprite: Objects/Tools/Hydroponics/diamondhoe.rsi
+
```
Comment (BombasterDS) [inline Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml:44]: - type: Construction
     graph: DiamondHoe
     node: hoe

**Comment by BombasterDS** [comment]
> > You need to add ConstructionComponent to it, as failed test says
> 
> I don't see ConstructionComponent in any YML prototypes

In YAML Component is skipping, so just search Construction

File: Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml [hunk 5]
```diff
@@ -19,6 +19,33 @@
...
+  - type: MeleeWeapon
+    wideAnimationRotation: 135
+    swingLeft: true
+    damage:
+      types:
+        Blunt: 10
+  - type: Item
+    sprite: Objects/Tools/Hydroponics/diamondhoe.rsi
+  - type: Item
+    sprite: Objects/Tools/Hydroponics/diamondhoe.rsi
```
Comment (Tayrtahn) [inline Resources/Prototypes/Entities/Objects/Specific/Hydroponics/tools.yml:48]: This somehow got duplicated

**Comment by BombasterDS** [comment]
And you better call nodes with lowercases. For example, you have DiamondHoe graph and DiamondHoe node. Better just call node hoe

**Comment by slarticodefast** [comment]
Are you coming back to this? If you need help fixing the errors, feel free to ping me on discord (Slartibartfast).

**Comment by slarticodefast** [comment]
If you plan to continue working on this, please write a comment and I will re-open the PR for you.

**Comment by Yeah-I-listening-Hollywood-undead** [comment]
> If you plan to continue working on this, please write a comment and I will re-open the PR for you.

Yes, I do. Everyone forget this PR and i think. Somewhere in the discord it was decided that this was not necessary

**Comment by slarticodefast** [comment]
The repository that submitted this pull request has been deleted, so I am unable to re-open it. Could you either restore it or make a new PR?

> Everyone forget this PR and i think. Somewhere in the discord it was decided that this was not necessary

There hasn't been an internal maintainer discussion thread for this yet, so I don't think such a decision has been made.
Personally I don't see a reason against including this, it's a fun little gimmick. You would have to address the reviews though.

**Comment by Yeah-I-listening-Hollywood-undead** [comment]
> There hasn't been an internal maintainer discussion thread for this yet, so I don't think such a decision has been made.
> Personally I don't see a reason against including this, it's a fun little gimmick. You would have to address the reviews and fix the test fails though before this can be merged.

Undersand. I am made new PR

**Pull request closed without merging**